src/components/views/rooms/AppsDrawer.js

```diff
@@ -219,9 +220,7 @@ module.exports = React.createClass({
             });
 
         let addWidget;
-        if (this.props.showApps &&
-            this._canUserModify()
-        ) {
+        if (this.props.showApps && this._canUserModify() && this.scalarClient) {
```

Comment:
I'd much prefer the entire `render` of the app draw to return a visible error instead of silently hiding the `addWidget` button